### PR TITLE
MSIZE opcode spec

### PR DESCRIPTION
**Pull request by spartucus**

### CODE REVIEW

File: specs/opcode/59MSIZE.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# MSIZE op code
+
+## Procedure
+
+The `MSIZE` opcode return the memory size of the current contract and place it on the top of the stack.
```
[the remark this answers is not in the document]
Comment (spartucus) [reply specs/opcode/59MSIZE.md:5]: > I think good to mention here is that the memory size returned by the opcode is in bytes, while internally the memory size is stored in words, so the opcode needs to multiply the state variable by `32`.

Yes, totally make sense.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,4 @@
 __pycache__
 *.egg-info
+
+build/
```
Comment (icemelon) [inline .gitignore:4]: add EOL

File: tests/test_msize.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+def test_msize():
+    memory = Memory()
+    check_msize(memory, 0)
+
+    memory.write(1, 0)
+    check_msize(memory, 1)
+    
+    memory.write(32, 0)
+    check_msize(memory, 2)
```
Comment (icemelon) [inline tests/test_msize.py:11]: add EOL

File: specs/opcode/59MSIZE.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+## Constraints
+
+1. opId = OpcodeId(0x59)
+2. state transition:
+    - gc + 1 (1 stack write)
+    - stack_pointer - 1
+    - pc + 1
+    - gas + 2
+3. lookups: `1` busmapping lookups
```
Comment (icemelon) [inline specs/opcode/59MSIZE.md:15]:
```suggestion
3. lookups: 1 busmapping lookup
```

File: specs/opcode/59MSIZE.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# MSIZE op code
+
+## Procedure
+
+The `MSIZE` opcode return the memory size of the current contract and place it on the top of the stack. The memory size returned by the opcode is in bytes, while internally the memory size is stored in words, so the opcode needs to multiply the state variable by `32`.
```
Comment (icemelon) [inline specs/opcode/59MSIZE.md:5]:
```suggestion
The `MSIZE` opcode returns the memory size of the current execution and places it on the top of the stack. The memory size returned by the opcode is in bytes, while internally the memory size is stored in words, so the opcode needs to multiply the state variable by 32.
```

File: specs/opcode/59MSIZE.md (hunk 4 above)
Comment (spartucus) [reply specs/opcode/59MSIZE.md]: done

**Comment by ChihChengLiang** [comment]
@spartucus What's the status of this PR? Anything you'd like to add before we merge this?

**Review by ChihChengLiang — APPROVED** [review]
LGTM

**Comment by spartucus** [comment]
> @spartucus What's the status of this PR? Anything you'd like to add before we merge this?

I just fixed conflicts, all good, ready to be merged :)

**Comment by ChihChengLiang** [comment]
@spartucus  Sorry, I added lint in the previous PR. Can you run `make fmt`?

**Comment by spartucus** [comment]
OK, done :)

**Pull request merged**